modules/server/src/topics/missiontopic.cpp

```diff
@@ -0,0 +1,124 @@
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2021                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/server/include/topics/missiontopic.h>
+
+#include <modules/server/include/connection.h>
+#include <modules/server/include/jsonconverters.h>
+#include <openspace/engine/globals.h>
+#include <openspace/mission/missionmanager.h>
+#include <openspace/util/time.h>
+#include <ghoul/logging/logmanager.h>
+#include <modules/spacecraftinstruments/util/imagesequencer.h>
+#include <openspace/util/spicemanager.h>
+
+using nlohmann::json;
+
+namespace openspace {
+
+bool MissionTopic::isDone() const {
+    return true;
+}
+
+nlohmann::json MissionTopic::missionJson() const {
+    const std::map<std::string, Mission>& missions =
+        global::missionManager->missionMap();
+
+    ImageSequencer& sequencer = ImageSequencer::ref();
+    const std::vector<double>& captureTimes = sequencer.captureProgression();
+    std::vector<std::string> captureTimesString(captureTimes.size());
+
+    for (int i = 0; i < captureTimes.size(); i++) {
+        const std::string& str = SpiceManager::ref().dateFromEphemerisTime(
```

Comment:
Better to do: `std::string str = Spice...`  and then `captureTimesString[i] = std::move(str)` since that is what is happening already, just less visibly so